client/core/core.go

```diff
@@ -2347,6 +2383,80 @@ func handleRevokeMatchMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) erro
 	return tracker.db.UpdateOrder(metaOrder)
 }
 
+// handleTradeSuspensionMsg is called when a trade suspension notification is received.
+func handleTradeSuspensionMsg(c *Core, dc *dexConnection, msg *msgjson.Message) error {
+	var sp msgjson.TradeSuspension
+	err := msg.Unmarshal(&sp)
+	if err != nil {
+		return fmt.Errorf("trade suspension unmarshal error: %v", err)
+	}
+
+	dc.pendingSuspendsMtx.Lock()
+	defer dc.pendingSuspendsMtx.Unlock()
+
+	// Attempt to cancel and remove pending suspends.
+	if sched := dc.pendingSuspends[sp.MarketID]; sched != nil {
+		if !sched.Stop() {
+			return fmt.Errorf("market %s for dex %s is already suspended",
+				sp.MarketID, dc.acct.host)
+		}
+		delete(dc.pendingSuspends, sp.MarketID)
+	}
+
+	// Set the new scheduled suspend.
+	duration := time.Until(encode.UnixTimeMilli(int64(sp.SuspendTime)))
+	dc.pendingSuspends[sp.MarketID] = time.AfterFunc(duration, func() {
+		dc.suspend(sp.MarketID)
```

Comment:
Sorry I didn't notice this earlier, but if this were to be called with a MarketID that is not in the maketMap, I think dc.marketMap[mkt].Suspended = true will cause a panic. Is that acceptable? perhaps a check as to whether the market exists in the map at the beginning of the function would be good? The check under the !sp.Persist conditional could just be moved up.

Reply:
Good catch, will update `suspend` and `resume` to ensure the market exists first and return an error if it doesn't, the check under `!sp.Persist` can then be removed.